python/artm/model.py

```diff
@@ -0,0 +1,583 @@
+            self._num_processors = num_processors
+
+    @num_document_passes.setter
+    def num_document_passes(self, num_document_passes):
+        if num_document_passes <= 0 or not isinstance(num_document_passes, int):
+            raise IOError("Number of passes through documents" +
+                          "should be a positive integer")
+        else:
+            self._num_document_passes = num_document_passes
+
+    @cache_theta.setter
+    def cache_theta(self, cache_theta):
+        if not isinstance(cache_theta, bool):
+            raise IOError('cache_theta should be bool')
+        else:
+            self.master.reconfigure(cache_theta=cache_theta)
+            self._cache_theta = cache_theta
+
+    @num_topics.setter
+    def num_topics(self, num_topics):
+        if num_topics <= 0 or not isinstance(num_topics, int):
+            raise IOError('Number of topics should be a positive integer')
+        else:
+            self._num_topics = num_topics
+
+    @topic_names.setter
+    def topic_names(self, topic_names):
+        if not topic_names:
+            raise IOError('Number of topic names should be non-negative')
+        else:
+            self._topic_names = topic_names
+            self._num_topics = len(topic_names)
+
+    @class_ids.setter
+    def class_ids(self, class_ids):
+        if len(class_ids) < 0:
+            raise IOError('Number of (class_id, class_weight) pairs should be non-negative')
+        else:
+            self._class_ids = class_ids
+
+    # ========== METHODS ==========
+    def load_dictionary(self, dictionary_name=None, dictionary_path=None):
+        """ARTM.load_dictionary() --- load the BigARTM dictionary of
+        the collection into the lib
+
+        Args:
+          dictionary_name (str): the name of the dictionary in the lib, default=None
+          dictionary_path (str): full file name of the dictionary, default=None
+        """
+        if dictionary_path is not None and dictionary_name is not None:
+            self.master.import_dictionary(file_name=dictionary_path,
+                                          dictionary_name=dictionary_name)
+        elif dictionary_path is None:
+            raise IOError('dictionary_path is None')
+        else:
+            raise IOError('dictionary_name is None')
+
+    def remove_dictionary(self, dictionary_name=None):
+        """ARTM.remove_dictionary() --- remove the loaded BigARTM dictionary
+        from the lib
+
+        Args:
+          dictionary_name (str): the name of the dictionary in th lib, default=None
+        """
+        if dictionary_name is not None:
+            self._lib.ArtmDisposeDictionary(self.master.master_id, dictionary_name)
+        else:
+            raise IOError('dictionary_name is None')
+
+    def fit_offline(self, batch_vectorizer=None, num_collection_passes=1, decay_weight=0.0,
+                    apply_weight=1.0, reset_theta_scores=False):
+        """ARTM.fit_offline() --- proceed the learning of
```

Comment:
It is not necessary to write the name of class and the method in the docstring, bacause you will easily see the whole signature in the documentation.
```python
In [3]: from sklearn import decomposition
In [5]: ?decomposition.NMF.fit
Signature: decomposition.NMF.fit(self, X, y=None, **params)
Docstring:
Learn a NMF model for the data X.
...
```